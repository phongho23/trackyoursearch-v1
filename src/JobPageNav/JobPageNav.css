.JobPageNav__back-button-wrapper {
  max-width: 380px;
  margin: 0 10px;
}

.JobPageNav__back-button {
  margin: 0 auto 20px;
}

.JobPageNav__back-button:hover {
  background-color: #553D67;
}

.JobPageNav__week-name {
color: #242582;
font-family: var(--custom-font);
padding-left: 5%;
padding-right: 10%;
padding-top: 5px;
display: flex;
padding: 5px 5px;
font-size: 32px;
}