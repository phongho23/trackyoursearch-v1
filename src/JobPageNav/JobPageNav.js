
import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import CircleButton from '../CircleButton/CircleButton'
import './JobPageNav.css'

export default function JobPageNav(props) {
  return (
    <div className='JobPageNav'>

      <div className='JobPageNav__back-button-wrapper'>
      <CircleButton
          tag='button'
          role='link'
          type='button'
          onClick={() => props.history.goBack()}
          className='JobPageNav__back-button'>
          <FontAwesomeIcon icon='chevron-left' />
          <br />
          Back
      </CircleButton>
      </div>

      {props.week && (
        <h3 className='JobPageNav__week-name'>
          {props.week.name}
        </h3>
      )}
    </div>
  )
}

JobPageNav.defaultProps = {
  history: {
    goBack: () => {}
  }
}
