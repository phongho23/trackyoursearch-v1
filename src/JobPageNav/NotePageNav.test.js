import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import JobPageNav from './JobPageNav'

describe(`JobPageNav component`, () => {
  const props = {
    week: {
      "name": "Important"
    }
  }

  it('renders a .JobPageNav by default', () => {
    const wrapper = shallow(<JobPageNav />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a h3 with week name when in props', () => {
    const h3 = shallow(<JobPageNav {...props} />)
      .find('.JobPageNav__week-name')
    expect(toJson(h3)).toMatchSnapshot()
  })
})
