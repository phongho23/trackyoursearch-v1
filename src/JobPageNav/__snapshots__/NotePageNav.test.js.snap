// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobPageNav component renders a .JobPageNav by default 1`] = `
<div
  className="JobPageNav"
>
  <NavCircleButton
    className="JobPageNav__back-button"
    onClick={[Function]}
    role="link"
    tag="button"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="chevron-left"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
    <br />
    Back
  </NavCircleButton>
</div>
`;

exports[`JobPageNav component renders a h3 with week name when in props 1`] = `
<h3
  className="JobPageNav__week-name"
>
  Important
</h3>
`;
